torchrl/modules/models/multiagent.py

```diff
@@ -229,6 +229,141 @@ def forward(self, *inputs: Tuple[torch.Tensor]) -> torch.Tensor:
+        kernel_sizes (int, Sequence[Union[int, Sequence[int]]]): Kernel size(s) of the conv network.
+            default: 5
+        strides (int or Sequence[int]): Stride(s) of the conv network. If iterable, the length must match the
+            depth, defined by the num_cells or depth arguments.
+            default: 2
+        activation_class (Type[nn.Module]): activation class to be used.
+            default: nn.ELU
+        **kwargs: for :class:`torchrl.modules.models.ConvNet` can be passed to customize the ConvNet.Args:
+
+    Examples:
+        >>> import torch
+        >>> from torchrl.modules import MultiAgentConvNet
+        >>> batch = (3,2)
+        >>> n_agents = 7
+        >>> x, y, channels = 100, 100, 3
+        >>> obs = torch.randn(*batch, n_agents, x, y, channels)
+        >>> cnn = MultiAgentConvNet(
+        >>>     n_agents,
+        >>>     centralised = True,
+        >>>     share_params = True
+        >>> )
+        >>> print(cnn)
+        MultiAgentConvNet(
+            (agent_networks): ModuleList(
+                (0): ConvNet(
+                (0): LazyConv2d(0, 32, kernel_size=(5, 5), stride=(2, 2))
+                (1): ELU(alpha=1.0)
+                (2): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (3): ELU(alpha=1.0)
+                (4): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (5): ELU(alpha=1.0)
+                (6): SquashDims()
+                )
+            )
+        )
+        >>> print(cnn(obs).shape)
+        torch.Size([3, 2, 7, 2592]) - where the final dimension is determiend by the input x,
+
+    """
+    def __init__(self,
+                 n_agents: int,
+                 centralised: bool,
+                 share_params: bool,
+                 device: Optional[DEVICE_TYPING] = None,
+                 num_cells: Optional[Sequence[int]] = None,
+                 kernel_sizes: Union[Sequence[Union[int, Sequence[int]]], int] = 5,
+                 strides: Union[Sequence, int] = 2,
+                 paddings: Union[Sequence, int] = 0,
+                 activation_class: Type[nn.Module] = nn.ELU,   
+                 **kwargs):
+        super().__init__()
+
+        self.n_agents = n_agents
+        self.centralised = centralised
+        self.share_params = share_params
+
+        self.agent_networks = nn.ModuleList(
+            [
+                ConvNet(
+                    num_cells=num_cells,
+                    kernel_sizes=kernel_sizes,
+                    strides=strides,
+                    paddings=paddings,
+                    activation_class=activation_class,
+                    device=device,
+                    **kwargs
+                )
+                for _ in range(self.n_agents if not self.share_params else 1)
+            ]
+        )
+
+
+    def forward(self, *inputs: Tuple[torch.Tensor]):
+        if len(inputs) > 1:
+            inputs = torch.cat([*inputs], -1)
+        else:
+            inputs = inputs[0]
```

Comment:
We should mimic the behavior of the ConvNet and expect only one input

Reply:
Removed